File: pinyougou_web_manager/src/main/webapp/js/service/brandService.js
//品牌服务
app.service("brandService",function ($http) {

    //查询所有
    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    }
    //分页查询
    this.findPage = function (page,size) {
        return $http.get("../brand/findPage.do?page=" + page + "&size=" + size);
    }
    //添加
    this.add = function (brand) {
        return $http.post("../brand/add.do", brand)
    }
    //修改
    this.update = function (brand) {
        return $http.post("../brand/update.do", brand)
    }
    //根据id查询实体
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    }
    //删除
    this.del = function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    }
    //条件查询
    this.search = function (page,size,searchBrand) {
        return $http.post("../brand/search.do?page=" + page + "&size=" + size,searchBrand);
    }
    //下拉列表数据
    this.selectOptionList = function () {
        return $http.get("../brand/selectOptionList.do");
    }
});

File: pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 */
public interface BrandService {
    /**
     * 查询所有品牌
     * @return
     */
    List<TbBrand> findAll();

    /**
     * 品牌分页
     * @param pageNum  当前页数
     * @param pageSize 每页的记录数
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);

    /**
     * 品牌添加操作
     * @param brand
     */
    void add(TbBrand brand);

    /**
     * 根据id查询实体
     * @param id
     * @return
     */
    TbBrand findOne(long id);

    /**
     * 修改
     * @param brand
     */
    void update(TbBrand brand);

    /**
     * 删除
     * @param ids
     */
    void delete(long[] ids);

    /**
     * 品牌条件查询
     * @param brand
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPage(TbBrand brand,int pageNum, int pageSize);

    /**
     * 返回下拉列表数据
     * @return
     */
    List<Map> selectOptionList();


}
